server/src/main/java/com/openelements/benchscape/server/store/repositories/EnvironmentRepository.java

```diff
@@ -19,14 +19,60 @@ default List<EnvironmentEntity> findAllByQuery(@NonNull final EnvironmentQuery e
+                predicates.add(criteriaBuilder.equal(root.get("gitOriginUrl"), environmentQuery.gitOriginUrl()));
+            }
+            if (environmentQuery.gitBranch() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("gitBranch"), environmentQuery.gitBranch()));
+            }
+            if (environmentQuery.systemProcessors() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemProcessors"), environmentQuery.systemProcessors()));
+            }
+            if (environmentQuery.systemProcessorsMin() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemProcessorsMin"), environmentQuery.systemProcessorsMin()));
+            }
+            if (environmentQuery.systemProcessorsMax() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemProcessorsMax"), environmentQuery.systemProcessorsMax()));
+            }
+            if (environmentQuery.systemMemory() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemMemory"), environmentQuery.systemMemory()));
+            }
+            if (environmentQuery.systemMemoryMin() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemMemoryMin"), environmentQuery.systemMemoryMin()));
+            }
+            if (environmentQuery.systemMemoryMax() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemMemoryMax"), environmentQuery.systemMemoryMax()));
+            }
+            if (environmentQuery.osName() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("osName"), environmentQuery.osName()));
+            }
+            if (environmentQuery.osVersion() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("osVersion"), environmentQuery.osVersion()));
+            }
+            if (environmentQuery.jvmVersion() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("jvmVersion"), environmentQuery.jvmVersion()));
+            }
+            if (environmentQuery.jvmName() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("jvmName"), environmentQuery.jvmName()));
+            }
+            if (environmentQuery.osFamily() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("osFamily"), environmentQuery.osFamily()));
+            }
+            if (environmentQuery.systemMemoryReadable() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("systemMemoryReadable"), environmentQuery.systemMemoryReadable()));
+            }
+            if (environmentQuery.cores() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("cores"), environmentQuery.cores()));
+            }
+            if (environmentQuery.jmhVersion() != null) {
+                predicates.add(criteriaBuilder.equal(root.get("jmhVersion"), environmentQuery.jmhVersion()));
```

Comment:
We should extract all the strings (like `"jmhVersion"` in this case) to constants in the class:

`private final static JMH_VERSION_QUERY_FIELD = "jmhVersion";
`
and than:

`                predicates.add(criteriaBuilder.equal(root.get(JMH_VERSION_QUERY_FIELD), environmentQuery.jmhVersion()));
`

I assume we will have more queries in future and by doing so we can reuse the field names.